### PR TITLE
feat(course): linking course page to backend

### PR DESCRIPTION
**Pull request by lucycy98**
### Purpose

creating a course in the frontend creates a course in the backend.
- updated course model to fit the required fields (see ticket)
- added error checking for courses (cannot leave fields empty) 
- makes a POST request (with authorisation) to create course

[HRT-60](https://softeng761team5.atlassian.net/browse/HRT-60)

### Does this PR introduce a breaking change?

- [] Yes
- [x] No

needed to update tests for creating a course in server.

#### Does this PR involve a UI change?

- [x] Yes
- [ ] No

#### Screenshot(s) of UI before this PR

image

#### Screenshot(s) of UI after this PR

image

### CODE REVIEW

**Review by EricPedrido — CHANGES_REQUESTED** [review]
Everything looks good :) I've noticed though that there's a number input for cohort size. It was like this before, but Claire said that she specifically wanted a more ballpark representation using ranges such as 
- Less than 20
- 20-50
- 50-100
- More than 100

Other than that everything works fine for me

**Comment by michellemtan** [comment]
Ignore this, I saw it matches the backend :) Looks good to me!

**Comment by gargi-bansal** [comment]
> Everything looks good :) I've noticed though that there's a number input for cohort size. It was like this before, but Claire said that she specifically wanted a more ballpark representation using ranges such as
> 
> * Less than 20
> * 20-50
> * 50-100
> * More than 100
> 
> Other than that everything works fine for me

@EricPedrido Is this something we should wait for her to respond to considering she’s on leave. The user could just input an estimate number in and it could be grouped on the backend for reporting purposes?

**Comment by EricPedrido** [comment]
> @EricPedrido Is this something we should wait for her to respond to considering she’s on leave. The user could just input an estimate number in and it could be grouped on the backend for reporting purposes?

@gargi-bansal It's something I remember her saying last time we had a meeting with her when I showed her the design for this page. She said it was because the specific number of people doesn't matter, it was more for the players to get a rough feel for the size of the class rather than an exact number. I think for now, your suggestion makes sense until we get the clarification from Claire though.

**Comment by preetpatel** [comment]
> > @EricPedrido Is this something we should wait for her to respond to considering she’s on leave. The user could just input an estimate number in and it could be grouped on the backend for reporting purposes?
> 
> @gargi-bansal It's something I remember her saying last time we had a meeting with her when I showed her the design for this page. She said it was because the specific number of people doesn't matter, it was more for the players to get a rough feel for the size of the class rather than an exact number. I think for now, your suggestion makes sense until we get the clarification from Claire though.

We're never going to get said clarification in time and I'm not too keen on adding an enum in the backend model so +1 on @gargi-bansal 's approach

File: server/course/course.model.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const courseSchemaDef = {
     type: String,
     required: true,
   },
-  ageOfCourse: {
+  startYear: {
```
Comment (preetpatel) [inline server/course/course.model.js]: remove these. They have been added in master

**Review by preetpatel — CHANGES_REQUESTED** [review]

**Review by preetpatel — APPROVED** [review]

**Review by EricPedrido — APPROVED** [review]
LGTM

**Review by TCHE614 — APPROVED** [review]

**Pull request merged**